### PR TITLE
fix(mock data): failed to cancel the mock data task

### PR DESCRIPTION
**Pull request by zijiacj**
#### What type of PR is this?
type-bug

#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/MockDataRuntimeFlowableTask.java [hunk 1]
```diff
@@ -61,7 +61,7 @@ public class MockDataRuntimeFlowableTask extends BaseODCFlowTaskDelegate<Void> {
 
     @Override
     public boolean cancel(boolean mayInterruptIfRunning, Long taskId, TaskService taskService) {
-        Verify.notNull(context, "MockContext is null");
+        Verify.notNull(context, "MockContext");
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/MockDataRuntimeFlowableTask.java:64]: throw an i18n exception

File: server/odc-core/src/main/resources/i18n/BusinessMessages_zh_TW.properties [hunk 2]
```diff
@@ -900,3 +900,6 @@ com.oceanbase.odc.PartitionPlanVariableKey.INTERVAL=間隔
 com.oceanbase.odc.PartitionPlanVariableKey.LAST_PARTITION_VALUE=最後一個分區規則對應位置表達式的值
 com.oceanbase.odc.partitionplan.TimeDataType=時間類型
 com.oceanbase.odc.partitionplan.NumberDataType=數字類型
+
+# MockData Task Message
+com.oceanbase.odc.MockData.null.message=未能獲取到{0}，任務還未開始執行或者已完成
```
Comment (yhilmare) [inline server/odc-core/src/main/resources/i18n/BusinessMessages_zh_TW.properties:905]: this is not a specific problem for mock data, we need to use a common error code to process this situation~

File: server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/MockDataRuntimeFlowableTask.java [hunk 3]
```diff
@@ -61,7 +65,14 @@ public class MockDataRuntimeFlowableTask extends BaseODCFlowTaskDelegate<Void> {
 
     @Override
     public boolean cancel(boolean mayInterruptIfRunning, Long taskId, TaskService taskService) {
-        Verify.notNull(context, "MockContext is null");
+        if (context == null) {
+            Locale locale = LocaleContextHolder.getLocale();
+            String i18nKey = "com.oceanbase.odc.MockData.null.message";
+            throw new VerifyException(I18n.translate(
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/MockDataRuntimeFlowableTask.java:71]: throw an http exception instead

File: server/odc-core/src/main/java/com/oceanbase/odc/core/shared/constant/ErrorCodes.java [hunk 4]
```diff
@@ -131,6 +131,7 @@ public enum ErrorCodes implements ErrorCode {
     TaskNotReadyForDownload,
     TaskSqlExecuteFailed,
     TaskLogNotFound,
+    TaskCannotCancelNow,
```
Comment (yhilmare) [inline server/odc-core/src/main/java/com/oceanbase/odc/core/shared/constant/ErrorCodes.java:134]: TaskFailedCancelledNow may be better?

File: server/odc-core/src/main/resources/i18n/ErrorMessages_zh_CN.properties [hunk 5]
```diff
@@ -195,3 +195,4 @@ com.oceanbase.odc.ErrorCodes.PartitionKeyDataTypeMismatch = 数据类型不匹
 com.oceanbase.odc.ErrorCodes.TimeDataTypePrecisionMismatch = 时间类型精度不匹配，配置的时间间隔精度大于分区键 {0} 数据类型的最大精度
 com.oceanbase.odc.ErrorCodes.DatabaseAccessDenied=数据库访问被拒绝，因为当前用户没有数据库的{0}权限
 com.oceanbase.odc.ErrorCodes.ObQueryProfileNotSupported=执行详情仅在高于或等于 {0} 的 OceanBase 版本可用
+com.oceanbase.odc.ErrorCodes.TaskCannotCancelNow=未能获取到{0}，任务还未开始执行或者已完成
```
Comment (yhilmare) [inline server/odc-core/src/main/resources/i18n/ErrorMessages_zh_CN.properties:198]: "当前无法取消，任务可能还未开始，请稍后再试" may be better?

File: server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/MockDataRuntimeFlowableTask.java [hunk 6]
```diff
@@ -61,7 +63,10 @@ public class MockDataRuntimeFlowableTask extends BaseODCFlowTaskDelegate<Void> {
 
     @Override
     public boolean cancel(boolean mayInterruptIfRunning, Long taskId, TaskService taskService) {
-        Verify.notNull(context, "MockContext is null");
+        if (context == null) {
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/MockDataRuntimeFlowableTask.java:66]: `new Object[] {}` will be fine, don't need to input any message

**Review by yhilmare — APPROVED** [review]
lgtm

**Review by Jane201510 — APPROVED** [review]

**Review by yhilmare — APPROVED** [review]
lgtm

**Review by yizhouxw — APPROVED** [review]

**Review by JessieWuJiexi — APPROVED** [review]

**Review by Jane201510 — APPROVED** [review]

**Pull request merged**